lib/cell.rb

```diff
@@ -23,7 +23,17 @@ def fired_upon?
   end
 
   def fire_upon
-    @ship.hit
+    if @ship != nil
+      @ship.hit
+    end
     @fired_upon = true
   end
+
+  def render(show_ship = false)
+    return "X" if @fired_upon == true && @ship.sunk? == true
```

Comment:
This is the one line I changed on the copy to get it working. 
Instead of @fired_upon == true, I changed it to @ship != nil and got a passing test.
I don't know why that one change affects it all!

Reply:
Awesome work. I am reviewing the other changes as well and will merge if all looks good!